src/core/forces_inline.hpp

```diff
@@ -192,15 +192,14 @@ inline Utils::Vector3d calc_non_bonded_pair_force_parts(
 #ifdef GAY_BERNE
   // The gb force function isn't inlined, probably due to its size
   if (dist < ia_params.gay_berne.cut) {
-    auto const forces =
-        gb_pair_force(p1.r.calc_director(), p2.r.calc_director(), ia_params, d,
-                      dist, torque1, torque2);
-    force += std::get<0>(forces);
+    auto const pf = gb_pair_force(p1.r.calc_director(), p2.r.calc_director(),
+                                  ia_params, d, dist, torque1);
+    force += pf.f;
     if (torque1) {
-      *torque1 += std::get<1>(forces);
+      *torque1 += pf.torque;
```

Comment:
I think there is a low hanging fruit here for considerable improvement: I think it would be better to return a `ParticleForce` from this function, instead of the sep. output parameters for the torques, and calculate the opposing
torque at the call site (as it is done for the force). If this is combined with a function `ParticleForce opposing_force(distance, f)` or similar, this can be made more ideomatic and there is no mention of torques anywhere is rotation is disabled. Also that function can be reused in other places, where this logic is also need.
If that isn't clear, the opposing force calculation is nothing specific for gay bern, it's just a consequence of of
momentum conservation, and this is the same for all central forces that produce toruqe. This is just a consequence
of Newtons law.

Reply:
done